### ISSUE TITLE
latest UEFI + ATF doesn't boot

### ISSUE THREAD

**Issue by koenkooi** [96boards/bugs#38]
```
Switch to aarch64 mode. CPU0 executes at 0xf9801000!
NOTICE:  Booting Trusted Firmware
NOTICE:  BL1: v1.1(release):be3e473
NOTICE:  BL1: Built : 16:27:51, Apr  3 2015
NOTICE:  acpu_dvfs_set_freq: set acpu freq success!NOTICE:  succeed to init lpddr3 rank0 dram phy
INFO:    lpddr3_freq_init, set ddrc 533mhz
INFO:    init ddr3 rank0 in 533MHz
INFO:    ddr3 rank1 init pass in 533MHz
INFO:    Elpida DDR
NOTICE:  BL1: Booting BL2
NOTICE:  BL2: v1.1(release):be3e473
NOTICE:  BL2: Built : 16:27:52, Apr  3 2015
ERROR:   Failed to load BL3-0 (-1)
PANIC at PC : 0x00000000f98194b0
```

edk2 rev: "766ab4ad50d80212a39a4bcc718782093122a72e"
atf rev: "be3e473d6f468ae8254a2b519eb0c1f9169f2232"

I flashed them using:

```
sudo python hisi-idt.py --img1=l-loader.bin
sudo fastboot flash ptable ptable.img
sudo fastboot flash fastboot fip.bin
```

**Comment by koenkooi** [comment]
old ATF + latest UEFI works, bisecting ATF now to narrow down the problem

**Comment by koenkooi** [comment]
BL30 is the mcuimage, making https://github.com/96boards/arm-trusted-firmware/commit/6d19f889ca559d75ecbaad490302e44a0a463217 the culprit

**Comment by koenkooi** [comment]
I still get the error after doing:

```
+ sudo fastboot flash mcuimage mcuimage.bin
target reported max download size of 134217728 bytes
sending 'mcuimage' (140 KB)...
OKAY [  0.003s]
writing 'mcuimage'...
OKAY [  0.014s]
finished. total time: 0.018s
```

The error:

```
INFO:    Elpida DDR
NOTICE:  BL1: Booting BL2
NOTICE:  BL2: v1.1(release):be3e473
NOTICE:  BL2: Built : 12:01:54, Apr  4 2015
ERROR:   Failed to load BL3-0 (-1)
PANIC at PC : 0x00000000f98194b0
```

Mcuimage is the one from the februari release:

```
[koen@thinkpad hisi-idt]$ md5sum mcuimage.bin 
aacb5268554fe2b61ac645d4e2e6d8da  mcuimage.bin
```

**Comment by vchong** [comment]
Can you try inserting mcuimage.bin into fip.bin as bl30?

In the arm-trusted-firmware directory, run:
tools/fip_create/fip_create fip.bin --dump --bl2 build/hikey/release/bl2.bin --bl31 build/hikey/release/bl31.bin --bl33 <path to BL33_AP_UEFI.fd, usually edk2/Build/HiKey/RELEASE_GCC49/FV/BL33_AP_UEFI.fd> --bl30 <path to mcuimage.bin>

Replace 'release' with 'debug' if necessary.

After that, sudo fastboot flash fastboot fip.bin

**Comment by Leo-Yan** [comment]
I have send email to Fathi to help change build script for that. So we need
update build script for mcu image loading.

Thanks,
Leo Yan
2015年4月4日 下午10:59于 "vchong" <notifications@github.com>写道：

**Comment by hzhuang1** [comment]
Now we fix it in pull request #23 first. Please check whether you still have this issue.

**Comment by koenkooi** [comment]
fixed in https://github.com/96boards/arm-trusted-firmware/pull/23
